kiali/kiali-ui
# [POC] Use cypress as proxy server for mock graph data

This change adds cypress.io to the project for the purpose of intercepting and mocking out the graph data response to return the pre-generated graph data created by the [generator tool](https://github.com/kiali/kiali/pull/4658).

To test this branch:
1. Update your deps to add cypress:  `yarn`
2. Update the `proxy` field in your `package.json` to point to your local kiali endpoint as per: https://github.com/kiali/kiali-ui#testing-with-kiali
3. ~Create namespaces `n[1-100]`. These namespaces need to exist for the graph page to work with the mock graph data.~ Run the [mock generator](https://github.com/kiali/kiali/pull/4658): `go run graph/generator/cmd/main.go`
4. Start your local front end server: `yarn start`
5. Start the cypress process: `yarn graph-proxy`
6. A cypress browser window should open. Click "Run 1 integration spec" in top right corner.
7. Navigate to the `graph` tab and select all the mock namespaces. You should see the mock graph data.

Something to note with using cypress as the proxy:

The request intercepting/proxy feature only works within the cypress browser context. If you navigate to the kiali url _outside_ of the cypress browser, you will not see the mock graph data but the real endpoint response. In other words, all testing of kiali with the mock data needs to be done within the context of the cypress browser.
